app/models/mixins/ansible_playbook_mixin.rb

```diff
@@ -35,9 +35,11 @@ def playbook_log_stdout(log_option, job)
     return unless %(on_error always).include?(log_option)
     return if log_option == 'on_error' && job.raw_status.succeeded?
 
-    $log.info("Stdout from ansible job #{job.name}: #{job.raw_stdout('txt_download')}")
+    $log.info("Stdout from ansible job #{job.name}: #{job.raw_stdout('txt_download')}") if job
```

Comment:
in the event we don't have a job, do we want to log that we're missing a job?

Reply:
@tinaafitz, @NickLaMuro, @billfitzgerald0120 What say you?

Reply:
who are the callers of this method and what is a reasonable expectation?

Reply:
Hi @chessbyte, 

This method gets called from the postprocess state.
https://github.com/ManageIQ/manageiq-content/blob/bb73854dd8d6b1e9647390538ef78046b123f608/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/postprocess.rb#L19 

The postprocess state is processed after the playbook has been launched.  A valid job is a reasonable expectation.

Reply:
See my comment here: https://github.com/ManageIQ/manageiq/pull/20738#issuecomment-719715815 .  IMO callers should be expected to pass a valid job.  I think the only thing that should change in this method is to add a guard clause:

```ruby
raise ArgumentError, "must pass a valid job" if job.nil?
```